src/plugins/kibana_usage_collection/server/collectors/application_usage/rollups.ts

```diff
@@ -58,10 +59,7 @@ export async function rollDailyData(logger: Logger, savedObjectsClient?: ISavedO
         } = doc;
         const dayId = moment(timestamp).format('YYYY-MM-DD');
 
-        const dailyId =
-          !viewId || viewId === MAIN_APP_DEFAULT_VIEW_ID
-            ? `${appId}:${dayId}`
-            : `${appId}:${dayId}:${viewId}`;
+        const dailyId = getDailyId({ dayId, appId, viewId });
```

Comment:
I did not delete `rollDailyData` or the `transactional` document usages in `src/plugins/kibana_usage_collection/server/collectors/application_usage/telemetry_application_usage_collector.ts`, as we're still waiting for the answer on https://github.com/elastic/kibana/issues/87840#issuecomment-801762769